source/tutorials/zfs.rst

```diff
@@ -0,0 +1,299 @@
+using a non-root device for your zpools. ZFS as a CLR_ROOT device
+is a not covered in this tutorial.
+
+.. contents::
+   :local:
+   :depth: 1
+
+Background
+**********
+
+Clear Linux does not ship with a binary ZFS kernel module (zfs.ko).
+ZFS is licensed under the CDDL. Linux is licensed under the GPL. These
+licenses are not compatible. Most Linux kernel developers assert that zfs.ko
+cannot be distributed in binary form alongside the kernel.
+
+Some distributions of Linux have a different opinion of
+CDDL / GPL compatibility -- most notably Canonical's Ubuntu -- and
+argue that a binary kernel module can be distributed alongside
+Linux without infringement. Clear Linux does not share that view.
+Therefore, zfs.ko must be built by the user under Clear Linux.
+
+It is worth understanding the `argument against shipping zfs.ko along
+with Linux`_ in addition to `the argument in favor`_.
+
+If you follow this tutorial and build zfs.ko on your system,
+*you should not redistribute that work in binary form*.
+
+Note that there is no issue with distributing ZFS in
+*source* form alongside Linux. According to those who argue
+both against and in favor of binary distribution, the CDDL and
+GPL are not incompatible with respect to redistribution of
+source code.
+
+There is nothing stopping a user from downloading the ZFS drivers,
+compiling them, then installing them for their own use. This is no
+different from downloading and using proprietary drivers like the Nvidia
+Graphics or Broadcom wireless drivers.
+
+Since the binaries for ZFS cannot be distributed with the
+Linux kernel binaries, you'll have some work to  ensure compatbility
+when new kernels are released.
+
+This is not always trivial, since new kernel release candidates will
+invariably break *something* in ZFS. The ZFS-on-Linux team is
+very good about catching up quickly, but on occasion you will find
+a lag between a new kernel release and a supported ZFS driver.
+
+Prerequisites
+*************
+
+To build ZFS, you need:
+
+DKMS kernel
+===========
+
+This tutorial assumes you are using a DKMS kernel. You should read and
+understand how to :ref:`kernel-modules-dkms`.
+
+If you do not currently use a DKMS kernel, install it with the following steps.
+
+Check whether you have an lts or native kernel:
+
+.. code-block:: bash
+
+    uname -r
+
+If 'native' appears in the kernel name, then install a native kernel with DKMS support:
+
+.. code-block:: bash
+
+    sudo swupd bundle-add kernel-native-dkms
+
+If 'lts' appears in the kernel name, then install the latest TLS kernel with DKMS support:
+
+.. code-block:: bash
+
+    sudo swupd bundle-add kernel-lts-dkms
+
+Reboot and make sure you can start the new kernel.
+
+Bundles
+=======
+
+Install these build tools before you install ZFS.
+
+If you are using a native kernel:
+
+.. code-block:: bash
+
+    sudo swupd bundle-add linux-dev
+
+If you are using an LTS kernel:
+
+.. code-block:: bash
+
+    sudo swupd bundle-add linux-lts-dev
+
```

Comment:
The `-dev` bundles are automatically included by the corresponding `-dkms` bundles, so they don't need to be added explicitly here.